### ISSUE TITLE
Output RTC/Jitify cache location / clearing the cache

### ISSUE THREAD

**Issue by ptheywood** [FLAMEGPU/FLAMEGPU2#454]
It's not immediately clear where the RTC cache is to the user, nor how to clear it. 

It's probably worth outputting the RTC cache location to stdout (if verbose?). 
It might also be worth providing a simple way to clear the cache.

**Comment by Robadob** [comment]
Whoops, I think I was supposed to create an issue re:auto/manual clearing of cache.

I don't think outputting it is necessary, it should just be documented alongside wherever the environment variable is documented (adding a note to the CMake option to disable it should suffice). It defaults to `/tmp/` which is perfectly sensible. The only complaint would be my choice to name the directory `fgpu2` rather than `flamegpu2`.

**Comment by ptheywood** [comment]
`/tmp` can be configured in many different ways though (which may not be clear to the user). 

I.e. on my office machine its `/tmp/user/1000/fgpu2/`, on my home machine its just `/tmp/fgpu2` (which would be an issue if multiple users tried to use it at the same time, mav is `/tmp/$USERNAME/fgpu2`, and HPC systems will again be different by default.

There's no harm in an extra `printf` to stdout (if verbose output is requested).

**Comment by Robadob** [comment]
If someone asks for verbose output, theyre expecting verbose with regards to their model execution. Not library internals. I think it just needs to be prominently mentioned in the rtc/python docs. Most software writes to /tmp/ it's normal behaviour.

**Comment by Robadob** [comment]
> on my home machine its just /tmp/fgpu2 (which would be an issue if multiple users tried to use it at the same time, 

And presumably its configured in that manner as it's a personal machine with 1 user....

**Comment by Robadob** [comment]
It's the same as the issue with the new CUDA profiler writing to TMP. Linux users with misconfigured systems unable to support multiple users.

**Comment by Robadob** [comment]
If you really want to support misconfigured linux, add `$user` to the folder hierarchy.

**Comment by Robadob** [comment]
Spoke with a couple of linux friends, both seemed to think it's upto the software to differentiate based on user. Was given the example that firefox creates temp directory `/tmp/mozilla-bader0` Where `bader` is presumably their username.

If that's to be applied, the vis resource tmp would need to be updated too.

**Comment by Robadob** [comment]
This change should do the trick, I basically added the `#if !defined(_MSC_VER)` block, nothing else should have changed. I expect it will be similar for Visualisation.
```C
path getTMP() {
    static path result;
    if (result.empty()) {
        path tmp = std::getenv("FLAMEGPU2_TMP_DIR") ? std::getenv("FLAMEGPU2_TMP_DIR") : temp_directory_path();
        // Create the $tmp/fgpu2/jitifycache(/debug) folder hierarchy
        if (!::exists(tmp) && !create_directory(tmp)) {
            THROW InvalidFilePath("Directory '%s' does not exist and cannot be created by JitifyCache.", tmp.generic_string().c_str());
        }
        if (!std::getenv("FLAMEGPU2_TMP_DIR")) {
            tmp /= "fgpu2";
#if !defined(_MSC_VER)
            // If linux tmp dir doesn't contain $user, add it to directory name
            const char *env_user = std::getenv("user") ? std::getenv("user") : "user";
            if (temp_directory_path().generic_string().find(env_user) == string::npos) {
                tmp += "-";
                tmp += env_user;
            }
#endif
            if (!::exists(tmp)) {
                create_directory(tmp);
            }
        }
        tmp /= "jitifycache";
        if (!::exists(tmp)) {
            create_directory(tmp);
        }
#ifdef _DEBUG
        tmp /= "debug";
        if (!::exists(tmp)) {
            create_directory(tmp);
        }
#endif
        result = tmp;
    }
    return result;
}
```

**Comment by Robadob** [comment]
The suggested change from the previous post has not been applied, I can sort it out if you're happy @ptheywood

**Comment by ptheywood** [comment]
the env var is generally `$USER` not `$user`, so that'll need changing (not sure if that's an actual standard or just convention). 

Lots of possible wierd corner cases too, i.e if the tmpdir is `/tmp` and the user is named `t, `mp`, `tmp` etc. 

Can also have per-user dirs, but based on the userid not the username, though cache sharing wouldn't be effected in that case, just the dir name would change.


If this change is made it will be more important to actually print the used temp dir to the user as well, can't just blanket say `rm -r $tmp/flamegpu` or similar, as it may or maynot be suffixed.

**Comment by ptheywood** [comment]
Closed by #999
